docs: add local storage support (#663)
harvester/docs: docs/advanced/addons/lvm-local-storage.md

![](/img/v1.4/csi-driver-lvm/create-lvm-sc-02.png)

3. Choose the target VG name. The VG name is the volume group name you created before.

![](/img/v1.4/csi-driver-lvm/create-lvm-sc-03.png)

4. Choose the target LV type. Currently we support **striped** and **dm-thin**.

![](/img/v1.4/csi-driver-lvm/create-lvm-sc-04.png)

5. Click **Save**. Then you can see the new StorageClass on the **Storage** page.

![](/img/v1.4/csi-driver-lvm/create-lvm-sc-05.png)

## Create Volume with LVM

You can create a new volume and attach it to the workload(VM). These steps are shown below:

1. Create a new volume with the new StorageClass you created before.

:::note

The volume status `Not Ready` is normal because the LVM volume will be created when the first workload is created.

:::

![](/img/v1.4/csi-driver-lvm/create-lvm-volume-01.png)

2. Attach the volume to the workload(VM).

:::note

The LVM volume is a local volume, so you need to ensure the SC target node is the same as the VM scheduled node.

:::

![](/img/v1.4/csi-driver-lvm/attach-lvm-volume-01.png)

3. After the volume attached, you can see the volume status is `Ready`.

> ![](/img/v1.4/csi-driver-lvm/attach-lvm-volume-02.png)

Author: Hi @jillian-maroket,
Because the volume is attached to the VM, the status is **In-use** instead **Ready**.

Thanks for correcting it.

![](/img/v1.4/csi-driver-lvm/create-vm-with-lvm-volume-01.png)

![](/img/v1.4/csi-driver-lvm/create-vm-with-lvm-volume-02.png)

## Config Snapshot/Backup for LVM volume

The LVM volume supports the snapshot and backup feature but you need to update the **csi-driver-config** in **Settings**. The steps are shown below:

1. Go to the **Settings** page and click the **Edit Setting** button on **csi-driver-config**.

![](/img/v1.4/csi-driver-lvm/update-csi-driver-config-01.png)

2. Add new **Provisioner**, **Volume Snapshot Class Name** and **Backup Volume Snapshot Class Name**.

![](/img/v1.4/csi-driver-lvm/update-csi-driver-config-02.png)

3. Now, you can create a new snapshot or backup for the LVM volume. (Here, snapshot is an example)

![](/img/v1.4/csi-driver-lvm/vm-take-snapshot-with-lvm-01.png)

4. check the snapshot status

![](/img/v1.4/csi-driver-lvm/vm-take-snapshot-with-lvm-02.png)

## Support features for LVM volume

:::caution

Although the backup feature looks good, it is not verified and supported.

:::

- **Snapshot**: Create a snapshot for the LVM volume.
- **Resize**: Resize the LVM volume.

> - **Clone**: Clone the LVM volume.

Reviewer: Why do we include backup creation in the list of supported features if it's not supported? Also, how are "verified" and "supported" different?

cc: @bk201 @asettle

> - **Clone**: Clone the LVM volume.

Author: We did not block the backup button in v1.4.0. And it is too late to add this change.

I discussed with Kiefer that it might not work because of the backup mechanism.
So, we need to add this warning to the document.

The verified and supported will be the same thing. We need to verify, and then we can say it is support.

> - **Clone**: Clone the LVM volume.

Reviewer 2: I suggest the term "not supported" to make it very clear.

## Creating a Volume with LVM

1. On the Harvester UI, go to the **Volumes** screen.

1. Create a new volume using the LVM StorageClass that you created.

  ![](/img/v1.4/csi-driver-lvm/create-lvm-volume-01.png)

  :::note

  The status **Not Ready** is normal because Harvester creates the LVM volume only when the first workload is created.

  :::

1. On the **Virtual Machines** screen, select the target virtual machine, and then select **⋮** > **Add Volume**.

  :::note

  Because the LVM volume is a local volume, you must ensure that the target node of the LVM StorageClass is the node on which the virtual machine is scheduled.

  :::

1. Specify the volume that you want to attach.

  ![](/img/v1.4/csi-driver-lvm/attach-lvm-volume-01.png)

1. On the **Volumes** screen, verify that the state is **In-use**.

  ![](/img/v1.4/csi-driver-lvm/attach-lvm-volume-02.png)

You can also create a new virtual machine with the volume of the LVM StorageClass that you created. This virtual machine will be scheduled on the target node with local storage for the volume.

![](/img/v1.4/csi-driver-lvm/create-vm-with-lvm-volume-01.png)

![](/img/v1.4/csi-driver-lvm/create-vm-with-lvm-volume-02.png)

> ## Creating Snapshots and Backups for an LVM Volume

Reviewer: ```suggestion
## Creating Snapshots for an LVM Volume
```

1. On the Harvester UI, go to the **Settings** screen.

1. In the **csi-driver-config** section, select **⋮** > **Edit Setting**.

  ![](/img/v1.4/csi-driver-lvm/update-csi-driver-config-01.png)

1. Add an entry with the following settings:

  - **Provisioner**: Select **lvm.driver.harvesterhci.io**.
  - **Volume Snapshot Class Name**: Select **lvm-snapshot**.
  - **Backup Volume Snapshot Class Name**: Select **lvm-snapshot**.

  ![](/img/v1.4/csi-driver-lvm/update-csi-driver-config-02.png)

> 1. On the **Virtual Machines** screen, select the target virtual machine, and then select **⋮** > **Take Backup** or **Take Virtual Machine Snapshot**.
> 
>   Example (snapshot):

Reviewer: ```suggestion
1. On the **Virtual Machines** screen, select the target virtual machine, and then select **⋮** > **Take Virtual Machine Snapshot**.

  Example:
```

  ![](/img/v1.4/csi-driver-lvm/vm-take-snapshot-with-lvm-01.png)

1. On the **Virtual Machine Snapshots** screen, verify that snapshot is ready to use.

  ![](/img/v1.4/csi-driver-lvm/vm-take-snapshot-with-lvm-02.png)

> ## Supported LVM Volume Features
> 
> - Volume resizing
> - Volume cloning
> - Snapshot creation
> - Backup creation (**Caution**: This feature is not fully verified and supported.)

Reviewer: ```suggestion
## Supported LVM Volume Features

- Volume resizing
- Volume cloning
- Snapshot creation

:::info important

Backup creation is currently not supported. This limitation will be addressed in a future release.

:::
```
